src/lib/distances.jl

```diff
@@ -79,11 +79,16 @@ end
 end
 
 @adjoint function pairwise(::Euclidean, X::AbstractMatrix; dims=2)
-  D, back = pullback(X -> pairwise(SqEuclidean(), X; dims = dims), X)
-  D .= sqrt.(D)
-  return D, function(Δ)
-    Δ = Δ ./ (2 .* max.(D, eps(eltype(D))))
-    Δ[diagind(Δ)] .= 0
-    return (nothing, first(back(Δ)))
+
+  _conditional(d, δ) = d > δ ? sqrt(d) : zero(δ)
```

Comment:
```suggestion
  _conditional(d, δ) = d > δ ? sqrt(d) : zero(d)
```
Shouldn't matter in this case as `d` and `δ` are the same type, but still to be as defensive as possible in case this code gets copied/reused/taken as an example.